Sources/SwiftBluetooth/CentralManager/CentralManager+async.swift

```diff
@@ -3,33 +3,50 @@ import CoreBluetooth
 
 public extension CentralManager {
     @available(iOS 13, macOS 10.15, watchOS 6.0, tvOS 13.0, *)
-    func waitUntilReady() async {
-        await withCheckedContinuation { cont in
-            self.waitUntilReady {
-                cont.resume()
+    func waitUntilReady() async throws {
+        try await withCheckedThrowingContinuation { cont in
+            self.waitUntilReady { result in
+                cont.resume(with: result)
             }
         }
     }
 
     @available(iOS 13, macOS 10.15, watchOS 6.0, tvOS 13.0, *)
     @discardableResult
-    func connect(_ peripheral: Peripheral, options: [String: Any]? = nil) async throws -> Peripheral {
-        try await withCheckedThrowingContinuation { cont in
-            self.connect(peripheral, options: options) { result in
-                switch result {
-                case .success(let peripheral):
-                    cont.resume(returning: peripheral)
-                case .failure(let error):
-                    cont.resume(throwing: error)
+    func connect(_ peripheral: Peripheral, timeout: TimeInterval, options: [String: Any]? = nil) async throws -> Peripheral {
+        var cancelled = false
+        var continuation: CheckedContinuation<Peripheral, Error>?
+        let cancel = {
+            cancelled = true
+            self.cancelPeripheralConnection(peripheral)
+            continuation?.resume(throwing: CentralError.cancelled)
```

Comment:
It may be better to throw a `CancellationError`: https://developer.apple.com/documentation/swift/cancellationerror